x-pack/plugin/src/test/resources/rest-api-spec/test/deprecation/10_basic.yml

```diff
@@ -10,7 +10,10 @@ setup:
       migration.deprecations:
         index: "*"
   - length: { cluster_settings: 0 }
-  - length: { node_settings: 0 }
+  # We don't check this field because for ES <8.0 && Java < 11, a node-level deprecation warning is triggered
+  # and it isn't easy to codify this in the REST tests without extra test infrastructure. `MigrationIT.java`
+  # does check this, however.
+  # - length: { node_settings: 0 }
```

Comment:
@rjernst can you can think of a better approach here?

Reply:
I think this is fine. Thanks for the note point to MigrationIT